### PR TITLE
v7-login: Fixed issues with login

### PR DESCRIPTION
**Pull request by saurjai3**
Fixed issues with login

### CODE REVIEW

File: packages/plugin-credentials/src/credentials/browser.js [hunk 1]
```diff
@@ -27,6 +27,11 @@ const Credentials = SparkPlugin.extend(Object.assign({}, common, {
     isLoggingIn: {
       default: false,
       type: `boolean`
+    },
+    // variable to check if there are user credentials with which auto login is possible, hence the client needs to wait until the credentails validation is completed.
```
Comment (ianwremmel) [inline packages/plugin-credentials/src/credentials/browser.js]: Wrap at 80 characters

File: packages/plugin-credentials/src/credentials/browser.js [hunk 2]
```diff
@@ -111,17 +116,17 @@ const Credentials = SparkPlugin.extend(Object.assign({}, common, {
...
+    }
 
+    process.nextTick(() => {
       if (query.code) {
-        Reflect.deleteProperty(location.query, `code`);
-        Reflect.deleteProperty(location.query, `state`);
-
-        this._updateLocation(location);
-
         // Though initialize is a synchronous call, it should be safe to
```
Comment (ianwremmel) [inline packages/plugin-credentials/src/credentials/browser.js]: I think these comments aren't valid anymore.

File: packages/plugin-credentials/src/credentials/browser.js [hunk 3]
```diff
@@ -147,6 +152,8 @@ const Credentials = SparkPlugin.extend(Object.assign({}, common, {
         this._updateLocation(location);
       }
 
+      // this should be set as false since we are no more validating the user for credentials. That means if the user already has credentials in his localStorage then he should be able to login without being presented the login screen.
```
Comment (ianwremmel) [inline packages/plugin-credentials/src/credentials/browser.js]: wrap comments at 80

File: packages/plugin-credentials/src/credentials/browser.js [hunk 4]
```diff
@@ -147,6 +152,8 @@ const Credentials = SparkPlugin.extend(Object.assign({}, common, {
...
+      this.isValidatingUser = false;
```
Comment (ianwremmel) [inline packages/plugin-credentials/src/credentials/browser.js]: Does this also need to happen in a `.then` block attached to `requestAuthorizationCodeGrant()` up above?

File: packages/plugin-credentials/src/credentials/browser.js [hunk 5]
```diff
@@ -27,6 +27,11 @@ const Credentials = SparkPlugin.extend(Object.assign({}, common, {
     isLoggingIn: {
       default: false,
       type: `boolean`
+    },
+    // variable to check if there are user credentials with which auto login is possible, hence the client needs to wait until the credentails validation is completed.
+    isValidatingUser: {
```
Comment (ianwremmel) [inline packages/plugin-credentials/src/credentials/browser.js]: How about flipping the states of this boolean and naming it something more like `queryStringProcessed`?

File: packages/plugin-credentials/src/credentials/browser.js (hunk 4 above)
Comment (saurjai3) [inline packages/plugin-credentials/src/credentials/browser.js]: No, this is the right location to reset the flag.

File: packages/plugin-credentials/src/index.js [hunk 6]
```diff
@@ -12,7 +12,9 @@ import config from './config';
 import AdvancedAuthInterceptor from './interceptor';
 
 const proxies = [
-  `canAuthorize`
+  `canAuthorize`,
+  `isAuthenticating`,
+  `queryStringProcessed`
```
Comment (ianwremmel) [inline packages/plugin-credentials/src/index.js]: Why does `queryStringProcessed` need to be proxied?

File: packages/plugin-credentials/src/credentials/common.js [hunk 7]
```diff
@@ -27,7 +27,7 @@ export default {
   session: {
     clientToken: makeStateDataType(Token, `token`).prop,
     isAuthenticating: {
-      default: false,
+      default: true,
```
Comment (ianwremmel) [inline packages/plugin-credentials/src/credentials/common.js]: If I'm reading this right, isAuthenticating will *always* be true in node environments.

File: packages/plugin-credentials/src/index.js (hunk 6 above)
Comment (saurjai3) [inline packages/plugin-credentials/src/index.js]: Needed to be proxied to let the web client use it as a notification to whether or not display the login page.

**Comment by saurjai3** [comment]
opened new PR: https://github.com/ciscospark/spark-js-sdk/pull/548
hence closing this one.

**Pull request closed without merging**